Build failure: lttng-tools

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- Try building with cross compilation
- Observe failure

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

https://hydra.nixos.org/build/300615877

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/f9jp4mxsr3j8dznqgnv10ps0d48wa57j-lttng-tools-2.13.15.tar.bz2
source root is lttng-tools-2.13.15
setting SOURCE_DATE_EPOCH to timestamp 1744398899 of file "lttng-tools-2.13.15/src/common/filter/filter-lexer.c"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./config/ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/840zd1vd2kqwsraq0xfyrsrnrvsancjf-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking target system type... aarch64-unknown-linux-gnu
checking for a BSD-compatible install... /nix/store/6sb7gak61cwcg3p5grnhhyndi5hz935y-coreutils-9.7/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/6sb7gak61cwcg3p5grnhhyndi5hz935y-coreutils-9.7/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) none
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... none
checking whether the C++ compiler works... yes
checking whether C compiler accepts -Wall... yes
checking whether C compiler accepts -Wno-incomplete-setjmp-declaration... yes
checking whether C compiler accepts -Wdiscarded-qualifiers... yes
checking whether C compiler accepts -Wmissing-declarations... yes
checking whether C compiler accepts -Wmissing-prototypes... yes
checking whether C compiler accepts -Wmissing-parameter-type... yes
checking whether C compiler accepts -Wold-style-definition... yes
checking whether C compiler accepts -Wstrict-prototypes... yes
checking whether C compiler accepts -Wshadow... yes
checking for grep that handles long lines and -e... /nix/store/kh0b6say9pqhlhv70sk2lxkgal6cs1xr-gnugrep-3.12/bin/grep
checking whether make sets $(MAKE)... (cached) yes
checking for a sed that does not truncate output... /nix/store/wmqv94vcaxp8zh877hinq9p1dqx3rzf6-gnused-4.9/bin/sed
checking for fold... /nix/store/6sb7gak61cwcg3p5grnhhyndi5hz935y-coreutils-9.7/bin/fold
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /nix/store/wmqv94vcaxp8zh877hinq9p1dqx3rzf6-gnused-4.9/bin/sed
checking for egrep... /nix/store/kh0b6say9pqhlhv70sk2lxkgal6cs1xr-gnugrep-3.12/bin/grep -E
checking for fgrep... /nix/store/kh0b6say9pqhlhv70sk2lxkgal6cs1xr-gnugrep-3.12/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert aarch64-unknown-linux-gnu file names to aarch64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert aarch64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/6sb7gak61cwcg3p5grnhhyndi5hz935y-coreutils-9.7/bin/dd
checking how to truncate binary pipes... /nix/store/6sb7gak61cwcg3p5grnhhyndi5hz935y-coreutils-9.7/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld
checking if the linker (ld) is GNU ld... yes
checking whether the g++ linker (ld) supports shared libraries... yes
/nix/store/kh0b6say9pqhlhv70sk2lxkgal6cs1xr-gnugrep-3.12/bin/grep: warning: stray \ before -
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for objcopy... objcopy
checking for pgrep... no
checking for bison... no
checking for byacc... no
checking for bison version... 
configure: WARNING:
Missing Bison >= 2.4. Note that the parser files are already built in
this distribution tarball, so Bison is only needed if you intend to
modify their sources. You can set the YACC variable to override automatic
detection.
    
checking for flex... no
checking for lex... no
checking for flex version... 
configure: WARNING:
Missing Flex >= 2.5.35. Note that the lexer files are already built in
this distribution tarball, so Flex is only needed if you intend to
modify their sources. You can set the LEX variable to override automatic
detection.
    
checking for inline... inline
checking for int32_t... yes
checking for int64_t... yes
checking for mode_t... yes
checking for off_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for ssize_t... yes
checking how to run the C preprocessor... gcc -E
checking for uid_t in sys/types.h... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for uint8_t... yes
checking for __attribute__... yes
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking whether the linker accepts -no-pie... yes
checking for library containing gethostbyname... none required
checking for library containing socket... none required
checking for occurrence(s) of link_all_deplibs = no in ./m4/libtool.m4... none
checking for sys/types.h... (cached) yes
checking for unistd.h... (cached) yes
checking for fcntl.h... yes
checking for string.h... (cached) yes
checking for pthread.h... yes
checking for limits.h... yes
checking for signal.h... yes
checking for stdlib.h... (cached) yes
checking for sys/un.h... yes
checking for sys/socket.h... yes
checking for stdlib.h... (cached) yes
checking for stdio.h... (cached) yes
checking for getopt.h... yes
checking for sys/ipc.h... yes
checking for sys/shm.h... yes
checking for popt.h... yes
checking for grp.h... yes
checking for arpa/inet.h... yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for paths.h... yes
checking for stddef.h... yes
checking for sys/file.h... yes
checking for sys/ioctl.h... yes
checking for sys/mount.h... yes
checking for sys/param.h... yes
checking for sys/time.h... yes
checking for elf.h... yes
checking for sys/random.h... yes
checking for sys/syscall.h... yes
checking for atexit... yes
checking for bzero... yes
checking for clock_gettime... yes
checking for dup2... yes
checking for fdatasync... yes
checking for fls... no
checking for ftruncate... yes
checking for gethostbyname... yes
checking for gethostname... yes
checking for getpagesize... yes
checking for localtime_r... yes
checking for memchr... yes
checking for memrchr... yes
checking for memset... yes
checking for mkdir... yes
checking for munmap... yes
checking for putenv... yes
checking for realpath... yes
checking for rmdir... yes
checking for socket... yes
checking for strchr... yes
checking for strcspn... yes
checking for strdup... yes
checking for strncasecmp... yes
checking for strndup... yes
checking for strnlen... yes
checking for strpbrk... yes
checking for strrchr... yes
checking for strstr... yes
checking for strtol... yes
checking for strtoul... yes
checking for strtoull... yes
checking for dirfd... yes
checking for gethostbyname2... yes
checking for getipnodebyname... no
checking for epoll_create1... yes
checking for sched_getcpu... yes
checking for sysconf... yes
checking for sync_file_range... yes
checking for getrandom... yes
checking for flock... yes
checking for pthread_setname_np(pthread_t, const char*)... yes
checking for pthread_setname_np(const char*)... no
checking for pthread_set_name_np(pthread_t, const char*)... no
checking for pthread_getname_np(pthread_t, char*, size_t)... yes
checking for pthread_getname_np(char*, size_t)... no
checking for pthread_get_name_np(pthread_t, char*, size_t)... no
checking for clock_gettime, timer_create, timer_settime, timer_delete in -lrt... yes
checking for dlopen in -ldl... yes
checking for dlmopen in -ldl... yes
checking pkg-config is at least version 0.9.0... yes
checking for popt... yes
checking for libxml-2.0 >= 2.7.6... yes
checking for clock_gettime... (cached) yes
checking for liburcu >= 0.11... yes
checking for liburcu-bp >= 0.11... yes
checking for liburcu-cds >= 0.11... yes
checking for kmod_module_probe_insert_module in -lkmod... yes
checking for lttng-ust >= 2.13... yes
checking for lttng-ust-ctl >= 2.13... yes
checking for fmemopen in -lc... yes
checking for pfm_initialize in -lpfm... no
configure: You may configure with --enable-python-bindings if you want Python bindings.
checking for linux/version.h... yes
checking for Linux epoll(7) interface... yes
checking for asciidoc... no
checking for xmlto... no
checking for python2... no
checking for python3... no
checking whether SDT probes compile... no
checking for dtrace... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating doc/examples/Makefile
config.status: creating doc/examples/rotation/Makefile
config.status: creating doc/examples/trigger-condition-event-matches/Makefile
config.status: creating doc/man/Makefile
config.status: creating doc/man/asciidoc-attrs.conf
config.status: creating include/Makefile
config.status: creating extras/Makefile
config.status: creating extras/bindings/Makefile
config.status: creating extras/bindings/swig/Makefile
config.status: creating extras/bindings/swig/python/Makefile
config.status: creating extras/core-handler/Makefile
config.status: creating src/Makefile
config.status: creating src/common/Makefile
config.status: creating src/common/argpar/Makefile
config.status: creating src/common/argpar-utils/Makefile
config.status: creating src/common/bytecode/Makefile
config.status: creating src/common/kernel-ctl/Makefile
config.status: creating src/common/kernel-consumer/Makefile
config.status: creating src/common/consumer/Makefile
config.status: creating src/common/ust-consumer/Makefile
config.status: creating src/common/hashtable/Makefile
config.status: creating src/common/sessiond-comm/Makefile
config.status: creating src/common/compat/Makefile
config.status: creating src/common/relayd/Makefile
config.status: creating src/common/testpoint/Makefile
config.status: creating src/common/index/Makefile
config.status: creating src/common/health/Makefile
config.status: creating src/common/config/Makefile
config.status: creating src/common/string-utils/Makefile
config.status: creating src/common/fd-tracker/Makefile
config.status: creating src/common/filter/Makefile
config.status: creating src/lib/Makefile
config.status: creating src/lib/lttng-ctl/Makefile
config.status: creating src/lib/lttng-ctl/lttng-ctl.pc
config.status: creating src/bin/Makefile
config.status: creating src/bin/lttng-consumerd/Makefile
config.status: creating src/bin/lttng-sessiond/Makefile
config.status: creating src/bin/lttng-relayd/Makefile
config.status: creating src/bin/lttng/Makefile
config.status: creating src/bin/lttng-crash/Makefile
config.status: creating src/vendor/Makefile
config.status: creating src/vendor/msgpack/Makefile
config.status: creating tests/Makefile
config.status: creating tests/destructive/Makefile
config.status: creating tests/regression/Makefile
config.status: creating tests/regression/kernel/Makefile
config.status: creating tests/regression/tools/Makefile
config.status: creating tests/regression/tools/streaming/Makefile
config.status: creating tests/regression/tools/filtering/Makefile
config.status: creating tests/regression/tools/health/Makefile
config.status: creating tests/regression/tools/tracefile-limits/Makefile
config.status: creating tests/regression/tools/snapshots/Makefile
config.status: creating tests/regression/tools/live/Makefile
config.status: creating tests/regression/tools/exclusion/Makefile
config.status: creating tests/regression/tools/save-load/Makefile
config.status: creating tests/regression/tools/save-load/configuration/Makefile
config.status: creating tests/regression/tools/mi/Makefile
config.status: creating tests/regression/tools/wildcard/Makefile
config.status: creating tests/regression/tools/channel/Makefile
config.status: creating tests/regression/tools/client/Makefile
config.status: creating tests/regression/tools/crash/Makefile
config.status: creating tests/regression/tools/regen-metadata/Makefile
config.status: creating tests/regression/tools/regen-statedump/Makefile
config.status: creating tests/regression/tools/notification/Makefile
config.status: creating tests/regression/tools/rotation/Makefile
config.status: creating tests/regression/tools/base-path/Makefile
config.status: creating tests/regression/tools/metadata/Makefile
config.status: creating tests/regression/tools/tracker/Makefile
config.status: creating tests/regression/tools/working-directory/Makefile
config.status: creating tests/regression/tools/relayd-grouping/Makefile
config.status: creating tests/regression/tools/clear/Makefile
config.status: creating tests/regression/tools/trigger/Makefile
config.status: creating tests/regression/tools/trigger/rate-policy/Makefile
config.status: creating tests/regression/tools/trigger/start-stop/Makefile
config.status: creating tests/regression/tools/trigger/utils/Makefile
config.status: creating tests/regression/tools/trigger/name/Makefile
config.status: creating tests/regression/tools/trigger/hidden/Makefile
config.status: creating tests/regression/ust/Makefile
config.status: creating tests/regression/ust/nprocesses/Makefile
config.status: creating tests/regression/ust/high-throughput/Makefile
config.status: creating tests/regression/ust/low-throughput/Makefile
config.status: creating tests/regression/ust/before-after/Makefile
config.status: creating tests/regression/ust/buffers-pid/Makefile
config.status: creating tests/regression/ust/periodical-metadata-flush/Makefile
config.status: creating tests/regression/ust/multi-session/Makefile
config.status: creating tests/regression/ust/multi-lib/Makefile
config.status: creating tests/regression/ust/overlap/Makefile
config.status: creating tests/regression/ust/overlap/demo/Makefile
config.status: creating tests/regression/ust/linking/Makefile
config.status: creating tests/regression/ust/daemon/Makefile
config.status: creating tests/regression/ust/exit-fast/Makefile
config.status: creating tests/regression/ust/fork/Makefile
config.status: creating tests/regression/ust/libc-wrapper/Makefile
config.status: creating tests/regression/ust/baddr-statedump/Makefile
config.status: creating tests/regression/ust/ust-dl/Makefile
config.status: creating tests/regression/ust/java-jul/Makefile
config.status: creating tests/regression/ust/java-log4j/Makefile
config.status: creating tests/regression/ust/java-log4j2/Makefile
config.status: creating tests/regression/ust/getcpu-override/Makefile
config.status: creating tests/regression/ust/clock-override/Makefile
config.status: creating tests/regression/ust/type-declarations/Makefile
config.status: creating tests/regression/ust/rotation-destroy-flush/Makefile
config.status: creating tests/regression/ust/blocking/Makefile
config.status: creating tests/regression/ust/namespaces/Makefile
config.status: creating tests/stress/Makefile
config.status: creating tests/unit/Makefile
config.status: creating tests/unit/ini_config/Makefile
config.status: creating tests/perf/Makefile
config.status: creating tests/utils/Makefile
config.status: creating tests/utils/tap/Makefile
config.status: creating tests/utils/testapp/Makefile
config.status: creating tests/utils/testapp/gen-data-pending/Makefile
config.status: creating tests/utils/testapp/gen-ns-events/Makefile
config.status: creating tests/utils/testapp/gen-kernel-test-events/Makefile
config.status: creating tests/utils/testapp/gen-py-events/Makefile
config.status: creating tests/utils/testapp/gen-ust-events/Makefile
config.status: creating tests/utils/testapp/gen-ust-events-ns/Makefile
config.status: creating tests/utils/testapp/gen-syscall-events-callstack/Makefile
config.status: creating tests/utils/testapp/gen-ust-nevents/Makefile
config.status: creating tests/utils/testapp/gen-ust-nevents-str/Makefile
config.status: creating tests/utils/testapp/gen-syscall-events/Makefile
config.status: creating tests/utils/testapp/gen-ust-tracef/Makefile
config.status: creating tests/utils/testapp/userspace-probe-elf-binary/Makefile
config.status: creating tests/utils/testapp/userspace-probe-elf-cxx-binary/Makefile
config.status: creating tests/utils/testapp/userspace-probe-sdt-binary/Makefile
config.status: creating tests/utils/xml-utils/Makefile
config.status: creating tests/regression/ust/python-logging/test_python_logging
config.status: creating tests/perf/test_perf_raw
config.status: creating tests/regression/ust/ust-dl/test_ust-dl
config.status: creating include/config.h
config.status: executing depfiles commands
config.status: executing libtool commands
checking for tput... no

LTTng-tools 2.13.15 "Nordicité"

The product of a collaboration between Champ Libre and Boréale, this farmhouse 
IPA is brewed with Kveik yeast and Québec-grown barley, oats and juniper 
branches. The result is a remarkable fruity hazy golden IPA that offers a 
balanced touch of resinous and woodsy bitterness.

Features
  Target architecture:                aarch64
  libkmod support:                    yes
  LTTng-UST support:                  yes

Binaries
  lttng:                              yes
  lttng-consumerd:                    yes
  lttng-crash:                        yes
  lttng-relayd:                       yes
  lttng-sessiond:                     yes

Extras:                               yes

Tests
  LTTng-UST Java agent JUL tests:     no
  LTTng-UST Java agent Log4j 1.x tests: no
  LTTng-UST Java agent Log4j 2.x tests: no
  LTTng-UST Python2 agent tests:      no
  LTTng-UST Python3 agent tests:      no
  LTTng-modules SDT uprobe tests:     no

Python binding:                       no
Build and install man pages:          yes (already built)
Embed --help messages:                no

Install directories
  Binaries:                           /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/bin
  Libraries:                          /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/lib

Search directories
  lttng-sessiond executable:          /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/bin/lttng-sessiond
  32-bit lttng-consumerd executable:  /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/lib/lttng/libexec/lttng-consumerd
  32-bit consumer libraries:          /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/lib
  64-bit lttng-consumerd executable:  /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/lib/lttng/libexec/lttng-consumerd
  64-bit consumer libraries:          /nix/store/b819r79hvv05ia46y9cjw5rk7a3m2gbv-lttng-tools-2.13.15/lib

WARNING: You need asciidoc and xmlto to build the LTTng-tools man pages.

Note that the man pages are already built in this distribution tarball,
therefore asciidoc and xmlto are only needed if you intend to modify
their sources.

Also note that the installed man pages will contain the project's
default command-line option and environment variable values.

Use --disable-man-pages to completely disable building and installing
the man pages.
configurePhase completed in 3 minutes 42 seconds
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/840zd1vd2kqwsraq0xfyrsrnrvsancjf-bash-5.2p37/bin/bash
Making all in include
make[1]: Entering directory '/build/lttng-tools-2.13.15/include'
make  all-am
make[2]: Entering directory '/build/lttng-tools-2.13.15/include'
  GEN        version.i
make[2]: Leaving directory '/build/lttng-tools-2.13.15/include'
make[1]: Leaving directory '/build/lttng-tools-2.13.15/include'
Making all in doc
make[1]: Entering directory '/build/lttng-tools-2.13.15/doc'
Making all in man
make[2]: Entering directory '/build/lttng-tools-2.13.15/doc/man'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/build/lttng-tools-2.13.15/doc/man'
make[2]: Entering directory '/build/lttng-tools-2.13.15/doc'
make[2]: Nothing to be done for 'all-am'.
make[2]: Leaving directory '/build/lttng-tools-2.13.15/doc'
make[1]: Leaving directory '/build/lttng-tools-2.13.15/doc'
Making all in src
make[1]: Entering directory '/build/lttng-tools-2.13.15/src'
Making all in vendor
make[2]: Entering directory '/build/lttng-tools-2.13.15/src/vendor'
Making all in msgpack
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/vendor/msgpack'
  CC       objectc.lo
  CC       unpack.lo
  CC       vrefbuffer.lo
  CC       zone.lo
  CC       version.lo
  CCLD     libmsgpack.la
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/vendor/msgpack'
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/vendor'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/vendor'
make[2]: Leaving directory '/build/lttng-tools-2.13.15/src/vendor'
Making all in common
make[2]: Entering directory '/build/lttng-tools-2.13.15/src/common'
Making all in argpar
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/common/argpar'
  CC       argpar.lo
  CCLD     libargpar.la
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/common/argpar'
Making all in bytecode
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/common/bytecode'
  CC       bytecode.lo
  CCLD     libbytecode.la
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/common/bytecode'
Making all in compat
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/common/compat'
  CC       poll.lo
  CC       compat-fcntl.lo
  CC       directory-handle.lo
  CCLD     libcompat.la
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/common/compat'
Making all in config
make[3]: Entering directory '/build/lttng-tools-2.13.15/src/common/config'
  CC       libconfig_la-ini.lo
  CC       libconfig_la-session-config.lo
[01m[Ksession-config.c:[m[K In function '[01m[Kencode_string[m[K':
[01m[Ksession-config.c:432:9:[m[K [01;35m[Kwarning: [m[K'[01m[Kinput[m[K' is deprecated [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  432 |         [01;35m[Kret[m[K = handler->input(out_str, &out_len, (const xmlChar *) in_str, &in_len);
      |         [01;35m[K^~~[m[K
In file included from [01m[K/nix/store/d3rbllv9zll9p3q3v5k20s0w6nm53z4g-libxml2-2.14.3-dev/include/libxml2/libxml/parser.h:25[m[K,
                 from [01m[Ksession-config.c:29[m[K:
[01m[K/nix/store/d3rbllv9zll9p3q3v5k20s0w6nm53z4g-libxml2-2.14.3-dev/include/libxml2/libxml/encoding.h:173:7:[m[K [01;36m[Knote: [m[Kdeclared here
  173 |     } [01;36m[Kinput[m[K XML_DEPRECATED_MEMBER;
      |       [01;36m[K^~~~~[m[K
[01m[Ksession-config.c:432:15:[m[K [01;31m[Kerror: [m[Kcalled object is not a function or function pointer
  432 |         ret = [01;31m[Khandler[m[K->input(out_str, &out_len, (const xmlChar *) in_str, &in_len);
      |               [01;31m[K^~~~~~~[m[K
At top level:
[01m[Kcc1:[m[K [01;36m[Knote: [m[Kunrecognized command-line option '[01m[K-Wno-incomplete-setjmp-declaration[m[K' may have been intended to silence earlier diagnostics
make[3]: *** [Makefile:635: libconfig_la-session-config.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/lttng-tools-2.13.15/src/common/config'
make[2]: *** [Makefile:1059: all-recursive] Error 1
make[2]: Leaving directory '/build/lttng-tools-2.13.15/src/common'
make[1]: *** [Makefile:553: all-recursive] Error 1
make[1]: Leaving directory '/build/lttng-tools-2.13.15/src'
make: *** [Makefile:664: all-recursive] Error 1
```

### Additional context

I'm not completely sure if the cross-compilation has anything to do with it, because I do not have a spare arm64 board laying around right now.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.3, NixOS, 25.11 (Xantusia), 25.11.20250627.30e2e28`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/igmrb3hj6wr27ll3nlgjfaawf5jsjvji-source`

### Notify maintainers

@bjornfor 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/lt/lttng-tools/package.nix
+++ b/pkgs/by-name/lt/lttng-tools/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchurl,
   pkg-config,
+  babeltrace2,
   popt,
   libuuid,
   liburcu,
@@ -13,15 +14,16 @@
 
 stdenv.mkDerivation rec {
   pname = "lttng-tools";
-  version = "2.13.15";
+  version = "2.14.0";
 
   src = fetchurl {
     url = "https://lttng.org/files/lttng-tools/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-lupCNR7hEsGdrZ/ceq6TtYPZ8XIrIXVmSjgdLTN3A8Q=";
+    sha256 = "sha256-2MOcJs7BO3vYJVHNUqIu/DWLiI4268+cG2DvHDo8L9M=";
   };
 
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [
+    babeltrace2
     popt
     libuuid
     liburcu

--- a/pkgs/by-name/lt/lttng-ust/package.nix
+++ b/pkgs/by-name/lt/lttng-ust/package.nix
@@ -26,13 +26,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "lttng-ust";
-  version = "2.13.9";
+  version = "2.14.0";
 
   src = fetchFromGitHub {
     owner = "lttng";
     repo = "lttng-ust";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-iwMhXUH569ey16C8F4HzU1ebZbn1oyDWOGpSIIokuGA=";
+    hash = "sha256-9WZDjOGfflEc6BUUO3W70KeLcZnTaePkF8eg8Ns/lQc=";
   };
 
   outputs = [

